nestkernel/node_collection.h

```diff
@@ -717,9 +718,34 @@ nc_const_iterator::operator+=( const size_t n )
   }
   else
   {
-    for ( size_t i = 0; i < n; ++i )
+    element_idx_ += n * step_;
+    // If we went past the size of the primitive, we need to adjust the element
+    // and primitive part indices.
+    size_t primitive_size = composite_collection_->parts_[ part_idx_ ].size();
+    while ( element_idx_ >= primitive_size )
     {
-      operator++();
+      element_idx_ = element_idx_ - primitive_size;
+      ++part_idx_;
+      if ( part_idx_ < composite_collection_->parts_.size() )
+      {
+        primitive_size = composite_collection_->parts_[ part_idx_ ].size();
+      }
+    }
+    // If we went past the end of the composite, we need to adjust the
+    // position of the iterator.
+    if ( composite_collection_->end_offset_ != 0 or composite_collection_->end_part_ != 0 )
+    {
+      if ( part_idx_ >= composite_collection_->end_part_ and element_idx_ >= composite_collection_->end_offset_ )
+      {
+        part_idx_ = composite_collection_->end_part_;
+        element_idx_ = composite_collection_->end_offset_;
+      }
+    }
+    else if ( part_idx_ >= composite_collection_->parts_.size() )
+    {
+      auto end_of_composite = composite_collection_->end();
+      part_idx_ = end_of_composite.part_idx_;
+      element_idx_ = end_of_composite.element_idx_;
```

Comment:
Except for the different steplength in line 721, this is a duplication of the code from `nc_const_iterator::operator++()`. I think it would therefore be better to put this code in a separate helper function (a private function within `nc_const_iterator`), perhaps with the steplength as argument, and call that function from both `operator++()` and `operator+=()`.

Reply:
That's true, it would make the code much nicer. Thanks! I will get to it in the next days.